src/util/psbt/serialize.rs

```diff
@@ -71,12 +76,36 @@ impl Serialize for PublicKey {
 }
 
 impl Deserialize for PublicKey {
+    #[inline]
     fn deserialize(bytes: &[u8]) -> Result<Self, encode::Error> {
         PublicKey::from_slice(bytes)
             .map_err(|_| encode::Error::ParseFailed("invalid public key"))
     }
 }
 
+impl Serialize for PartialSignature {
+    fn serialize(&self) -> Vec<u8> {
+        let mut buf = self.signature.serialize_der().to_vec();
+        buf.push(self.sighash_type as u8);
+        buf
+    }
+}
+
+impl Deserialize for PartialSignature {
+    fn deserialize(bytes: &[u8]) -> Result<Self, encode::Error> {
+        let (sig_slice, sighash_byte) = match bytes.len() {
+            70..=73 => (&bytes[..bytes.len() - 1], bytes[bytes.len() - 1]),
```

Comment:
Fun fact: It is also possible to have short signatures(< 70 bytes) in bitcoin. See:
bitcoin/bitcoin@ea5a50f/src/secp256k1/src/ecdsa_impl.h#L189-L190

I think with a probability of 1/1024 we can have sigs of 70 bytes. 1/(1024*256) 69 bytes and so on. Russell o'connor from Blockstream has used sigs as short as `300602010102010101` on testnet.